P2-issue-disposal.md

```diff
@@ -0,0 +1,57 @@
+Github Issue Disposal Policy
+-------------------------------------------------------
+* Author(s): makdharma
+* Approver: a11r
+* Status: Draft
+* Implemented in: N/A
+* Last updated: 2017-06-15
+* Discussion at: N/A
+
+## Abstract
+
+There are hundreds of open issues in the 3 gRPC repos, some open for over 2 years. gRPC team periodically spends valuable time scrubbing the issue list. This proposal lays down concrete criteria for disposing of issues.
+
+## Background
+
+The github Issues database serves many purposes.
+
+* Capture bugs
+* Keep track of TODO
+* Submit feature requests
+* Ask questions (ad hoc documentation)
+
+The team uses labels to informally classify issues. The default process is to keep the issue open unless we explicitly take care of all the work related with the issue. This process has an inherent tradeoff.
+
+* Each open issue consumes some engineering bandwidth in periodic bug scrubs
+* Leaving issue open adds uncertainty
+* Closing an issue loses potentially important information
+* The disposition process is not consistent across repos and users
+
+Can we do better?
+
+## Proposal
+
+We propose the following.
+
+### Keep status updated
+
+Close following types of issues with explanation. Anyone can reopen if they disagree with the disposition.
+* Issues that are fixed, or should have been fixed
+* Issues that are waiting for more information and have not seen an update in over a month
+
+### Correct classification
+
+Issues that start out as bugs can turn into feature requests or document requests. Use following *mutually exclusive* labels to identify the nature of an issue - Question, Enhancement, Bug.
+
+### Bring transparency to feature request handling
+
+* Old feature requests (pending for over a year)
+	* Make an honest assessment about feasibility and utility. Reject if appropriate with explanation. If the community feels strongly about the feature, anyone can open a new issue with added justification. This issue can then follow the New Issue path below.
+	* If the feature request makes sense but we have no immediate plans, then consolidate in a separate document and close the issue. This is equivalent of adding "Someday" label.
```

Comment:
Could this be expanded more? I don't get a strong grasp of how this would look.

Given how easy it is to filter issues with `Someday...` _milestone_ (or the Java-equivalent of Unscheduled milestone), what is this providing? It seems currently there are 100 open `Someday...` issues, and actually 97 are closed. Taking a small sampling of 6ish closed issues showed all were resolved, so maybe removing them from the issue tracker isn't a good idea, or maybe not everyone is using `Someday...` as strongly as would be done here.

Interestingly, Java is in a worse place here than grpc/grpc with 97 open Unscheduled issues.